met-web/src/components/engagement/form/EngagementWidgets/ContactInfoPaper.tsx

```diff
@@ -1,55 +1,160 @@
+            // When dragging upwards, only move when the cursor is above 50%
+
+            // Dragging downwards
+            if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+                return;
+            }
+
+            // Dragging upwards
+            if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+                return;
+            }
+
+            // Time to actually perform the action
+            moveContact(dragIndex, hoverIndex);
+
+            // Note: we're mutating the monitor item here!
+            // Generally it's better to avoid mutations,
+            // but it's good here for the sake of performance
+            // to avoid expensive index searches.
+            item.index = hoverIndex;
+        },
+    });
+
+    const [{ isDragging }, drag] = useDrag({
+        type: 'Contact',
+        item: () => {
+            return { testId, index };
+        },
+        collect: (monitor: any) => ({
+            isDragging: monitor.isDragging(),
+        }),
+    });
+
+    drag(drop(ref));
+
     return (
-        <MetWidgetPaper elevation={3} {...rest}>
+        <MetWidgetPaper ref={ref} elevation={3} {...rest}>
             <Grid container direction="row" alignItems={'flex-start'} justifyContent="flex-start">
-                <Grid item xs={2} container direction="row" alignItems={'flex-start'} justifyContent="flex-start">
+                <Grid item xs={1}>
+                    <IconButton sx={{ padding: 0, margin: 0 }} color="info" aria-label="drag-indicator">
+                        <DragIndicatorIcon />
+                    </IconButton>
+                </Grid>
+                <Grid item xs={3} container direction="row" alignItems={'flex-start'} justifyContent="flex-start">
                     <Grid item xs={12}>
                         <MetLabel>{contact.name}</MetLabel>
                     </Grid>
                     <Grid item xs={12}>
-                        <MetParagraph>{contact.title}</MetParagraph>
+                        <MetParagraph overflow="hidden" textOverflow={'ellipsis'} whiteSpace="nowrap">
+                            {contact.title}
+                        </MetParagraph>
                     </Grid>
                 </Grid>
                 <Grid
                     item
-                    xs={10}
+                    xs={6.5}
                     container
                     direction="row"
                     alignItems={'flex-start'}
                     justifyContent="flex-start"
-                    spacing={2}
+                    spacing={1}
                 >
-                    <Grid item xs={2}>
-                        <MetParagraph>Phone:</MetParagraph>
-                    </Grid>
-                    <Grid item xs={10}>
-                        <MetParagraph>{contact.phone_number}</MetParagraph>
-                    </Grid>
+                    <If condition={contact.phone_number}>
```

Comment:
Not entirely sure, but If requires to have an <Else> if the condition was falsey. The <While> Tag doesn't, It just renders children if condition is truthy so maybe use that?

Reply:
The While* tag